File: mobile/lib/classes/history.dart
class EnrichedGameHistory {
  final String date;
  final String startingTime;
  final String duration;
  final String gameMode;
  final List<String> playersId;
  final List<String> playersUsername;
  final String? winnerId;
  final List<String>? rageQuittersId;
  final List<MapEntry<String, int>>? differenceCounts;

  EnrichedGameHistory({
    required this.date,
    required this.startingTime,
    required this.duration,
    required this.gameMode,
    required this.playersId,
    required this.playersUsername,
    this.winnerId,
    this.rageQuittersId,
    this.differenceCounts,
  });

  factory EnrichedGameHistory.fromJson(Map<String, dynamic> json) {
    return EnrichedGameHistory(
      date: json['date'],
      startingTime: json['startingTime'],
      duration: json['duration'],
      gameMode: json['gameMode'],
      playersId: List<String>.from(json['playersId']),
      playersUsername: List<String>.from(json['playersUsername']),
      winnerId: json['winnerId'],
      rageQuittersId: json['rageQuittersId'] != null
          ? List<String>.from(json['rageQuittersId'])
          : null,
      differenceCounts: json['differenceCounts'] != null
          ? (json['differenceCounts'] as List)
              .map((item) => MapEntry(item[0] as String, item[1] as int))
              .toList()
          : null,
    );
  }

  @override
  String toString() {
    return 'EnrichedGameHistory{date: $date, startingTime: $startingTime, duration: $duration, gameMode: $gameMode, playersId: $playersId, playersUsername: $playersUsername, winnerId: $winnerId, rageQuittersId: $rageQuittersId, differenceCounts: $differenceCounts}';
  }
}

File: common/constants/history.ts
export const GAMES_HISTORY_STUB = [
    {
        date: new Date().toLocaleDateString('en-GB'),
        startingTime: '16:21:36',
        duration: '01:00',
        gameMode: 'Mode temps limité coop',
        playersId: ['1', '2', '3'],
        winnerId: '1',
        rageQuittersId: [],
        differenceCounts: [
            ['1', 5],
            ['2', 3],
            ['3', 2],
        ],
    },
    {
        date: new Date().toLocaleDateString('en-GB'),
        startingTime: '08:15:40',
        duration: '02:00',
        gameMode: 'Mode classique duo',
        playersId: ['1', '2'],
        winnerId: '2',
        rageQuittersId: [],
        differenceCounts: [
            ['1', 3],
            ['2', 5],
        ],
    },
];
